models/intermediate/int_asana__task_projects.sql

```diff
@@ -1,57 +1,87 @@
 with task_project as (
 
-    select * 
+    select *
     from {{ var('project_task') }}
 
 ),
 
 project as (
-    
-    select * 
+
+    select *
     from {{ var ('project') }}
 ),
 
 task_section as (
 
-    select * 
+    select *
     from {{ var('task_section') }}
 
 ),
 
 section as (
-    
-    select * 
+
+    select *
     from {{ var ('section') }}
 
 ),
 
 task_project_section as (
 
-    select 
+    select
         task_project.task_id,
         project.project_name || (case when section.section_name = '(no section)' then ''
-            else ': ' || section.section_name end) as project_section, 
-        project.project_id
+            else ': ' || section.section_name end) as project_section,
+        cast(project.project_id as {{ dbt.type_string() }}) as project_id,
+        project.project_name,
+        -- Using the first detected task project pairing as the "primary"
+        row_number() over (partition by task_project.task_id 
+            order by task_project._fivetran_synced asc, project.created_at asc) = 1 as is_first_project
     from
-    task_project 
-    join project 
+    task_project
+    join project
         on project.project_id = task_project.project_id
     join task_section
         on task_section.task_id = task_project.task_id
-    join section 
-        on section.section_id = task_section.section_id 
+    join section
+        on section.section_id = task_section.section_id
         and section.project_id = project.project_id
 ),
 
+task_project_primary as (
+    select distinct
+        task_id,
+        task_project_section.project_id as first_project_id,
+        task_project_section.project_name as first_project_name
+    from task_project_section
+    where is_first_project
+),
+
 agg_project_sections as (
-    select 
+    select
         task_id,
         {{ fivetran_utils.string_agg( 'task_project_section.project_section', "', '" )}} as projects_sections,
+        {{ fivetran_utils.string_agg( 'task_project_section.project_id', "', '" )}} as project_ids,
+        {{ fivetran_utils.string_agg( 'task_project_section.project_name', "', '" )}} as project_names,
```

Comment:
@irvingpop makes a great callout that leveraging these fields can be a bit strange and difficult to work with out of the box. I would include in the definition of the fields (within the asana.yml) that these fields will include *all* ids/projects the ticket is associated with and may be parsed or exploded out to better analyze. However, for the purpose of keeping the integrity of the grain of the table, we chose to not parse/explode these arrays out by default.

This way we can provide a bit more direction towards how to leverage these new fields best. It would be great to also include this blurb in the CHANGELOG so customers upgrading are aware when inspecting the release notes.

Reply:
Done!